contracts/GatewayComposer.sol

```diff
@@ -0,0 +1,374 @@
+     *
+     *      As per requirement bounty token currency is same as valueToken.
+     *      Bounty flow:
+     *          - facilitator approves GC for base tokens as bounty
+     *          - if bounty is greater than 0, it's transferred to GC
+     *          - GC approves Gateway for the bounty
+     *
+     * @param _stakeRequestHash Unique hash for each stake request.
+     * @param _r R of the signature.
+     * @param _s S of the signature.
+     * @param _v V of the signature.
+     * @param _hashLock Hash lock provided by the facilitator.
+     *
+     * @return messageHash_ Message hash unique for each stake request.
+     */
+    function acceptStakeRequest(
+        bytes32 _stakeRequestHash,
+        bytes32 _r,
+        bytes32 _s,
+        uint8 _v,
+        bytes32 _hashLock
+    )
+        external
+        returns (bytes32 messageHash_)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(
+            stakeRequest.stakeVT > uint256(0),
+            "Stake request not found."
+        );
+
+        uint256 bounty = GatewayInterface(stakeRequest.gateway).bounty();
+        valueToken.transferFrom(msg.sender, address(this), bounty);
+        valueToken.approve(stakeRequest.gateway, bounty);
+
+        require(
+            BrandedToken(brandedToken).acceptStakeRequest(
+                _stakeRequestHash,
+                _r,
+                _s,
+                _v
+            ),
+            "BrandedToken acceptStakeRequest returned false."
+        );
+
+        uint256 mintBT = BrandedToken(brandedToken).convertToBrandedTokens(
+            stakeRequest.stakeVT
+        );
+
+        valueToken.approve(stakeRequest.gateway, mintBT);
+
+        messageHash_ = GatewayInterface(stakeRequest.gateway).stake(
+            mintBT,
+            stakeRequest.beneficiary,
+            stakeRequest.gasPrice,
+            stakeRequest.gasLimit,
+            stakeRequest.nonce,
+            _hashLock
+        );
+
+        delete stakeRequests[_stakeRequestHash];
+    }
+
+    /**
+     * @notice Revokes stake request by calling BT.revokeStakeRequest() and
+     *         deleting information.
+     *
+     * @dev Function requires:
+     *          - stake request hash is valid
+     *          - BT.revokeStakeRequest() should return true
+     *
+     * @param _stakeRequestHash Stake request hash.
+     *
+     * @return success_ True on successful execution.
+     */
+    function revokeStakeRequest(
+        bytes32 _stakeRequestHash
+    )
+        external
+        onlyOwner
+        returns (bool success_)
+    {
+        StakeRequest storage stakeRequest = stakeRequests[_stakeRequestHash];
+        require(
+            stakeRequest.stakeVT > uint256(0),
+            "Stake request not found."
+        );
+        require(
+            BrandedToken(brandedToken).revokeStakeRequest(_stakeRequestHash),
+            "BrandedToken revokeStakeRequest returned false."
+        );
+
```

Comment:
we can `valueToken.transfer(owner, stakeVT)` after successfully redeeming (ie. GC is now has a valueToken balance